### PR TITLE
snowflake-snowpark-python 1.8.0 for py311 :snowflake:

### PR DESCRIPTION
**Pull request by skupr-anaconda**
**Current package info**

| Channel | Downloads | Version | Platforms | License |
| --- | --- | --- | --- | --- |
| main | [Conda Downloads](https://anaconda.org/main/snowflake-snowpark-python) | [Conda Version](https://anaconda.org/main/snowflake-snowpark-python) | [Conda Platforms](https://anaconda.org/main/snowflake-snowpark-python) | Conda License |
| sfe1ed40 | [Conda Downloads](https://anaconda.org/sfe1ed40/snowflake-snowpark-python) | [Conda Version](https://anaconda.org/sfe1ed40/snowflake-snowpark-python) | [Conda Platforms](https://anaconda.org/sfe1ed40/snowflake-snowpark-python) | Conda License |

Changelog: https://github.com/snowflakedb/snowpark-python/releases
License: https://github.com/snowflakedb/snowpark-python/blob/v1.8.0/LICENSE.txt
Requirements: https://github.com/snowflakedb/snowpark-python/blob/v1.8.0/setup.py

Actions:
1.   Skip `[(not py==311) or s390x]`
2. Add  `script_env`: `SNOWFLAKE_IS_PYTHON_RUNTIME_TEST=1` to run a build for **py311** support
3. Remove `pinning` for `setuptools` in `host`
4. Update pinning in `run`: `snowflake-connector-python >=3.2.0,<4.0.0`
5. Use `cloudpickle ==2.2.1` for `py311`

Notes:
- We build **py311** for the `sfe1ed40` channel
- For py311 there is an opened PR https://github.com/AnacondaRecipes/snowpark-python-feedstock/pull/31

### CODE REVIEW

File: abs.yaml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+upload_channels:
+  - sfe1ed40
```
Comment (M-Waszkiewicz-Anaconda) [inline abs.yaml:2]: the non-py.311 version goes to defaults, only py3.11 version goes to snowflake

**Review by M-Waszkiewicz-Anaconda — CHANGES_REQUESTED** [review]

**Review by M-Waszkiewicz-Anaconda — APPROVED** [review]
LGTM

**Comment by skupr-anaconda** [comment]
@M-Waszkiewicz-Anaconda FYI, [snowflake-ml-python](https://github.com/AnacondaRecipes/snowflake-ml-python-feedstock/pull/3) relies on this PR for enabling py311 support

**Review by varlackc — APPROVED** [review]
Looks Good

**Pull request merged**